### PR TITLE
Fix doc warnings in xilem_core, xilem_web

### PR DESCRIPTION
**Pull request by waywardmonkeys**

### CODE REVIEW

**Review by jaredoconnell — APPROVED** [review]

**Review by DJMcNab — APPROVED** [review]
Can we automate this being checked?

That is, how did you determine these errors?

It's ok to add lints to lib.rs if they are crate specific, so long as we document that in the root Cargo.toml

**Comment by waywardmonkeys** [comment]
Sure, we do in every other repo. This is blocked by #449 in theory, but I've decided to take a different approach to this today once this PR another lands.

**Pull request merged**
